wgengine/router/router_linux.go

```diff
@@ -718,23 +698,30 @@ func (r *linuxRouter) addNetfilterBase() error {
 		return fmt.Errorf("adding %v in filter/ts-input: %w", args, err)
 	}
 
-	// Forward and mark packets that have the Tailscale subnet route
-	// bit set. The bit gets set by rules inserted into filter/FORWARD
-	// later on. We use packet marks here so both filter/FORWARD and
-	// nat/POSTROUTING can match on these packets of interest.
+	// Forward all traffic from the Tailscale interface, and drop
+	// traffic to the tailscale interface by default. We use packet
+	// marks here so both filter/FORWARD and nat/POSTROUTING can match
+	// on these packets of interest.
 	//
 	// In particular, we only want to apply SNAT rules in
 	// nat/POSTROUTING to packets that originated from the Tailscale
 	// interface, but we can't match on the inbound interface in
-	// POSTROUTING. So instead, we match on the inbound interface and
-	// destination IP in filter/FORWARD, and set a packet mark that
-	// nat/POSTROUTING can use to effectively run that same test
-	// again.
+	// POSTROUTING. So instead, we match on the inbound interface in
+	// filter/FORWARD, and set a packet mark that nat/POSTROUTING can
+	// use to effectively run that same test again.
+	args = []string{"-i", r.tunname, "-j", "MARK", "--set-mark", tailscaleSubnetRouteMark}
+	if err := r.ipt4.Append("filter", "ts-forward", args...); err != nil {
+		return fmt.Errorf("adding %v in filter/ts-forward: %w", args, err)
+	}
 	args = []string{"-m", "mark", "--mark", tailscaleSubnetRouteMark, "-j", "ACCEPT"}
 	if err := r.ipt4.Append("filter", "ts-forward", args...); err != nil {
 		return fmt.Errorf("adding %v in filter/ts-forward: %w", args, err)
 	}
-	args = []string{"-i", r.tunname, "-j", "DROP"}
+	args = []string{"-o", r.tunname, "-s", "100.64.0.0/10", "-j", "DROP"}
```

Comment:
Should this be disallowed? It seems like in a non-masquerading setup, where tailscaled is running on the LAN's default gateway for example, it's entirely reasonable for LAN node 192.168.1.5 to forward through us to 100.1.2.3. Or will that traffic already have the tailscaleSubnetRouteMark?

Reply:
You're reading it backwards, I think. This rule says: if we're trying to forward a packet _to_ tailscale0 _from_ 100.64.0.0/10, drop it, because we should only ever be originating tailscale IP traffic from the current node, not somewhere else.

Reply:
Ah, rats, I saw '-o' and my brain then filled in '-d' instead of '-s'. Yes, this makes sense. Good!